pilot/pkg/features/pilot.go

```diff
@@ -48,6 +48,13 @@ var (
 		"Limits the number of concurrent pushes allowed. On larger machines this can be increased for faster pushes",
 	).Get()
 
+	// MaxRecvMsgSize The max receive buffer size of gRPC received channel of Pilot in bytes.
+	MaxRecvMsgSize = env.RegisterIntVar(
+		"ISTIO_GPRC_MAXRECVMSGSIZE",
```

Comment:
which grpc connection? we already have a similar flag https://github.com/istio/istio/blob/master/pilot/cmd/pilot-discovery/main.go#L131

Reply:
The xDS connection between the Proxy and the client. The parameter you said is used for MCP connection between Pilot and galley.

Reply:
The xDS connection between the Proxy and the client. The argument you said is used for MCP connection between Pilot and galley.

Reply:
can we make this xds_ instead of istio_ then so it is clear?

does envoy envoy have max size?

Reply:
The default send size is unlimited

Reply:
@howardjohn The name is consistent with `ISTIO_GPRC_MAXSTREAMS`.

Reply:
fair enough

Reply:
This is the option on Pilot server side and doesn't have any side effect on Envoy. If we name this "xds_xxxxxx", it will lead to misunderstanding because it looks like to change the receive buffer size of all xDS connections including the MCP.

On the other hand, in order to make this parameter more target precisely, we can change this "PILOT_SERVER_GPRC_MAXRECVMSGSIZE".